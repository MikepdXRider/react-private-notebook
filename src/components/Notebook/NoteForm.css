.form {
  display: grid;
  gap: 1rem;
  border: 2px dashed var(--gray);
  width: 450px;
}

.formSection {
  display: grid;
  gap: 0.25rem;
}

.input {
  padding: 7px 10px;
  border: 1px solid var(--gray);
  font-size: 1.1rem;
  font-family: ProximaNova;
}
