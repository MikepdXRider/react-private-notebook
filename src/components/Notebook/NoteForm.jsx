import { useEffect, useState } from 'react';

import styles from './NoteForm.css';

export default function NoteForm({
  formLabel = 'Note Form',
  title = '',
  content = '',
  className = '',
  onSubmit,
}) {
  const [formState, setFormState] = useState({ title, content });

  useEffect(() => {
    if (title || content) setFormState({ title, content });
  }, [title, content]);

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormState((prevState) => ({ ...prevState, [name]: value }));
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    await onSubmit(formState);
  };

  return (
    <form className={className} onSubmit={handleSubmit}>
      <fieldset className={styles.form}>
        <legend>{formLabel}</legend>
        <section className={styles.formSection}>
          <label htmlFor="title">Title</label>
          <input
            id="title"
            name="title"
            type="text"
            value={formState.title}
            className={styles.input}
            onChange={handleChange}
          />
        </section>
        <section className={styles.formSection}>
          <label htmlFor="content">Body</label>
          <textarea
            id="content"
            name="content"
            value={formState.content}
            className={styles.input}
            rows={7}
            onChange={handleChange}
          />
        </section>
        <button type="submit">Save</button>
      </fieldset>
    </form>
  );
}
