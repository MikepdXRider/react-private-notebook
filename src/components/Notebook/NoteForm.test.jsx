import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NoteForm from './NoteForm';

it('should display an empty form when no props provided', () => {
  const { container } = render(
    <MemoryRouter>
      <NoteForm />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

it('should display a form using the props provided as values', () => {
  const note = {
    title: 'Test Note',
    content: 'This is my note',
  };
  const mockSubmit = jest.fn();

  const { container } = render(
    <MemoryRouter>
      <NoteForm
        formLabel="Test Note Form"
        className="my-class-name"
        onSubmit={mockSubmit}
        {...note}
      />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();

  fireEvent.click(screen.getByRole('button'));
  expect(mockSubmit).toBeCalledTimes(1);
});
