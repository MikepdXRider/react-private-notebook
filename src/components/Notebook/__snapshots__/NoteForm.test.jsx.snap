// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a form using the props provided as values 1`] = `
<div>
  <form
    class="my-class-name"
  >
    <fieldset
      class="form"
    >
      <legend>
        Test Note Form
      </legend>
      <section
        class="formSection"
      >
        <label
          for="title"
        >
          Title
        </label>
        <input
          class="input"
          id="title"
          name="title"
          type="text"
          value="Test Note"
        />
      </section>
      <section
        class="formSection"
      >
        <label
          for="content"
        >
          Body
        </label>
        <textarea
          class="input"
          id="content"
          name="content"
          rows="7"
        >
          This is my note
        </textarea>
      </section>
      <button
        type="submit"
      >
        Save
      </button>
    </fieldset>
  </form>
</div>
`;

exports[`should display an empty form when no props provided 1`] = `
<div>
  <form
    class=""
  >
    <fieldset
      class="form"
    >
      <legend>
        Note Form
      </legend>
      <section
        class="formSection"
      >
        <label
          for="title"
        >
          Title
        </label>
        <input
          class="input"
          id="title"
          name="title"
          type="text"
          value=""
        />
      </section>
      <section
        class="formSection"
      >
        <label
          for="content"
        >
          Body
        </label>
        <textarea
          class="input"
          id="content"
          name="content"
          rows="7"
        />
      </section>
      <button
        type="submit"
      >
        Save
      </button>
    </fieldset>
  </form>
</div>
`;
